docs/Development/General/Refactoring_3.md

```diff
@@ -22,11 +27,41 @@ Add the following capability for defining the (default) admin icon:
 ```
 
 You can remove the old `admin.png` file afterwards.
+In addtion, it is recommended that you increment your module version one full point (i.e. 3.0.0 -> 4.0.0). Upgrading to core 3.0 
+is not backward compatible with core 2.0. Also, change the php version to >=7.2.5 (or higher if you module requires it) and the 
+core-compatibility to >=3.0.0.
+```{
+  ...
+  "version": "5.0.0",
+  ...
+  "require": {
+    "php": ">7.2.5",
+    "zikula/core-bundle": "3.*"
+  },
+  "extra": {
+    "zikula": {
+      ...
+      "core-compatibility": ">=3.0.0",
+      ...
+      "icon": "fas fa-star",
+      ...
+    }
+  }
+}
+  ```
 
 ### Interfaces
 
-In general, interfaces and apis implement argument type-hinting in all methods. This can break an implementation of said
-interfaces, etc. Extensions must update their implementation of any core/system interface to adhere to the new signature.
+In general, interfaces and apis implement argument type-hinting in all methods. This can break an implementation of said interfaces, etc. Extensions must update their implementation of any core/system interface to adhere to the new signature. For example, if you are upgrading a Block class, a member function line goes from:
```

Comment:
breaking lines makes them easier to read in the IDE or when viewing with a text-based application. They are rendered together when the markdown is converted. So this is preferred. I recommend breaking this up again.